src/models/marker.js

```diff
@@ -41,7 +41,28 @@ const Marker = Model.extend({
   },
 
   setSpace(newSpace) {
-    this.space = this._root.dimensionManager.getDimensionModelsForSpace(this._space, newSpace);
+    const setProps = {};
+    const newDimModels = setProps["space"] = this._root.dimensionManager.getDimensionModelsForSpace(this._space, newSpace);
+    const addedDimModels = newDimModels.filter(f => !this.space.includes(f));
+    addedDimModels.forEach(dimensionModel => {
+      const dimModel = this.getClosestModel(dimensionModel);
+      const labelModelName = "label" + dimensionModel.replace(dimModel._type, "");
+      setProps[labelModelName] = { which: dimModel.dim, use: "property", spaceRef: dimensionModel };
+      let founded = true;
+      //change which to 'name' if 'name' property available for dimension
+      utils.forEach(this._root._data, dataSource => {
```

Comment:
marker model should not know about internal workings of data models
instead, create a function in data manager (or use one if there is already), which tells you if a certain key-value pair is available. Nowhere in code above the data manager should we care about data sources. If we need country names here we shouldn´t care which source they would come from